src/ComputerVirtualMachine.php

```diff
@@ -434,6 +434,13 @@ public static function findVirtualMachine($fields = [])
             $result = $iterator->current();
             return $result['id'];
         } else if (count($iterator) > 1) {
+            for ($i = 0; $i < count($iterator); $i++) {
+                $iterator->seek($i);
+                if ($iterator->current()['uuid'] == $fields['uuid']) {
+                    $result = $iterator->current();
+                    return $result['id'];
+                }
+            }
```

Comment:
If we want to be sure that we have only one match, we should do something like this.

```suggestion
            $exact_matches = array_filter(
                iterator_to_array($iterator),
                static function (array $row) use ($fields) {
                    return strtolower($row['uuid']) === strtolower($fields['uuid']);
                }
            );
            if (count($exact_matches) === 1) {
                return end($exact_matches)['id'];
            }
```